streamalert_cli/outputs/handler.py

```diff
@@ -13,43 +13,60 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
+import json
+
 from streamalert.shared.logger import get_logger
 from streamalert.alert_processor.outputs.output_base import (
     StreamAlertOutput,
     OutputCredentialsProvider
 )
-from streamalert_cli.helpers import user_input
+from streamalert_cli.helpers import user_input, response_is_valid
 from streamalert_cli.outputs.helpers import output_exists
-from streamalert_cli.utils import CLICommand
+from streamalert_cli.utils import CLICommand, generate_subparser
 
 LOGGER = get_logger(__name__)
+OUTPUTS_FILE = 'outputs_to_configure.json'
 
 
 class OutputCommand(CLICommand):
-    description = 'Create a new StreamAlert output'
+    description = 'Describe and manage StreamAlert outputs'
 
     @classmethod
     def setup_subparser(cls, subparser):
-        """Add the output subparser: manage.py output SERVICE"""
-        outputs = sorted(StreamAlertOutput.get_all_outputs().keys())
+        """Add the output subparser: manage.py output"""
+        output_subparsers = subparser.add_subparsers()
 
-        # Output parser arguments
-        subparser.add_argument(
-            'service',
-            choices=outputs,
-            metavar='SERVICE',
-            help='Create a new StreamAlert output for one of the available services: {}'.format(
-                ', '.join(outputs)
-            )
-        )
+        for subcommand in cls._subcommands().values():
+            subcommand.setup_subparser(output_subparsers)
 
     @classmethod
     def handler(cls, options, config):
-        """Configure a new output for this service
+        subcommands = cls._subcommands()
 
-        Args:
-            options (argparse.Namespace): Basically a namedtuple with the service setting
+        if options.subcommand in subcommands:
+            return subcommands[options.subcommand].handler(options, config)
+
+        LOGGER.error('Unhandled output subcommand %s', options.subcommand)
+
+    @staticmethod
+    def _subcommands():
+        return {
+            'set': OutputSetSubCommand,
+            'set-from-file': OutputSetFromFileSubCommand,
+            'generate-skeleton': OutputGenerateSkeletonSubCommand,
+            'get': OutputGetSubCommand
```

Comment:
It would be helpful if you can add `list` command to list current existing outputs. Just output names. So users don't need to dig into outputs.json file.

Reply:
True! Will add :D